src/Core/MySQL/MySQLReplication.cpp

```diff
@@ -562,6 +563,13 @@ namespace MySQLReplication
                         row.push_back(dispatch((meta >> 8) & 0xFF, meta & 0xFF, read_decimal));
                         break;
                     }
+                    case MYSQL_TYPE_ENUM:
+                    {
+                        UInt8 val = 0;
+                        payload.readStrict(reinterpret_cast<char *>(&val), 1);
```

Comment:
The payload may be one or two bytes here, depending on the enum. `meta & 0xFF` should give you the number of bytes.